### ISSUE TITLE
`npm install --no-optional` fails on sockit-to-me

### ISSUE THREAD

**Issue by chrmod** [mozilla/browser.html#668]
Tried to install dependencies for current master (same for 0.0.8), with following node versions:
* v0.11.11
* v4.0.0
* iojs-v1.6.4

OS is Ubuntu 14.10.

That's the error message:
```bash
npm http 200 https://registry.npmjs.org/sockit-to-me/-/sockit-to-me-0.3.2.tgz

> sockit-to-me@0.3.2 install /home/chrmod/Projects/opensource/browser.html/node_modules/marionette-client/node_modules/sockit-to-me
> ./tools/copy.js || node-gyp configure build

[sockit-to-me] found g++
make: Entering directory '/home/chrmod/Projects/opensource/browser.html/node_modules/marionette-client/node_modules/sockit-to-me/build'
make: *** No rule to make target 'Release/obj.target/sockit/src/node-0.11/addon.o', needed by 'Release/obj.target/sockit.node'.  Stop.
make: Leaving directory '/home/chrmod/Projects/opensource/browser.html/node_modules/marionette-client/node_modules/sockit-to-me/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/chrmod/.nvm/v0.11.11/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:107:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:879:12)
gyp ERR! System Linux 3.16.0-44-generic
gyp ERR! command "node" "/home/chrmod/.nvm/v0.11.11/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build"
gyp ERR! cwd /home/chrmod/Projects/opensource/browser.html/node_modules/marionette-client/node_modules/sockit-to-me
gyp ERR! node -v v0.11.11
gyp ERR! node-gyp -v v0.12.2
gyp ERR! not ok 
npm ERR! sockit-to-me@0.3.2 install: `./tools/copy.js || node-gyp configure build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sockit-to-me@0.3.2 install script.
npm ERR! This is most likely a problem with the sockit-to-me package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./tools/copy.js || node-gyp configure build
npm ERR! You can get their info via:
npm ERR!     npm owner ls sockit-to-me
npm ERR! There is likely additional logging output above.

npm ERR! System Linux 3.16.0-44-generic
npm ERR! command "/home/chrmod/.nvm/v0.11.11/bin/node" "/home/chrmod/.nvm/v0.11.11/bin/npm" "install" "--no-optional"
npm ERR! cwd /home/chrmod/Projects/opensource/browser.html
npm ERR! node -v v0.11.11
npm ERR! npm -v 1.3.25
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/chrmod/Projects/opensource/browser.html/npm-debug.log
npm ERR! not ok code 0
```

What is the suggested build environment for current master?

**Comment by Gozala** [comment]
Have you tried with current master there was some dep versions issue that was fixed by this change
https://github.com/mozilla/browser.html/commit/d16b7060c0049257a7deee2222c3eae07a604b05

**Comment by chrmod** [comment]
Got it. **sockit-to-me** does only support node version 0.10 and 0.12.

**Comment by KevinGrandon** [comment]
A fix for node 4.0 and sockit-to-me is in progress here: https://bugzilla.mozilla.org/show_bug.cgi?id=1202921

**Comment by KevinGrandon** [comment]
This should work now with node4+.
